### PR TITLE
New Hardhat & Solidity versions

### LINKED ISSUE #524: Update Solidity pragma version and Hardhat in Semaphore V4

### PR DESCRIPTION
**Pull request by cedoor**
## Related Issue

Closes #524 

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: packages/contracts/contracts/Semaphore.sol [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.4;
+pragma solidity 0.8.19;
```
Comment (vplasencia) [inline packages/contracts/contracts/Semaphore.sol]: Why not use `0.8.23` instead of `0.8.19`?

Comment (cedoor) [reply packages/contracts/contracts/Semaphore.sol]: It's the one Hardhat sets when you init a project. I think it's better to use it for compatibility. There are some warnings with the latest.

Comment (vplasencia) [reply packages/contracts/contracts/Semaphore.sol]: Which warnings? I tested with `0.8.23` and I didn't see any warnings.

Comment (cedoor) [reply packages/contracts/contracts/Semaphore.sol]: You're right :D I've probably seen them with Viem or some other configuration. Let me update it

**Review by vplasencia — APPROVED** [review]

**Pull request merged**